src/components/side_panel/find_and_replace/find_and_replace.ts

```diff
@@ -101,7 +111,6 @@ export class FindAndReplacePanel extends Component<Props, SpreadsheetChildEnv> {
 
   onFocusSidePanel() {
     this.state.searchOptions.searchFormulas = this.env.model.getters.shouldShowFormulas();
-    this.env.model.dispatch("REFRESH_SEARCH");
```

Comment:
Previously, there was an issue where, in "All sheets" mode, clicking on the side panel after manually changing the active sheet would take you back to the previous sheet.

However consider the case where we are on the 3rd search of Sheet 1 and click on the side panel. This will refresh the search and move us back to the first search.

Reply:
Well then we introduce another issue - 
Open find & replace, start a search
go to top bar > view > show formula
refocus the side panel
the 'search in formula' boolean is ticked but the search is not updated so I don't think it's the proper fix.

the problem is that we'd refresh the search at every focus whereas we should have only done it when the value of "shouldShowFormulas" was updated. That being said, I alseo think that reacting on 'focus' was a bad idea fro the start and we should rather react when the 'showFormula' is updated for the whole sheet.
this can be achieved with a useEffect that spies on `this.env.model.getters.shouldShowFormulas`

Reply:
I think the method onfocussidepanel has become obsolete  now